### PR TITLE
Add IO utilities

### PR DESCRIPTION
**Pull request by ebocher**
- to import, export a file in a database (H2GIS, POSTGIS)
- to export a table from another database (H2GIS, POSTGIS)
- to link a file or table (H2GIS only)

### CODE REVIEW

File: h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IOMethods.class);
+
+    private static final String ENCODING_OPTION = "charset=";
+    private static final String UTF_ENCODING = "UTF-8";
+
+    private static DriverFunction getDriverFromFile(File file) {
+        String path = file.getAbsolutePath();
+        String extension = "";
+        int i = path.lastIndexOf(46);
```
Comment (SPalominos) [inline h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java:65]: Maybe `lastIndexOf('.');` is more readable.

File: h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+import org.h2gis.utilities.FileUtilities;
+import org.h2gis.utilities.JDBCUtilities;
+import org.h2gis.utilities.TableLocation;
+import org.h2gis.utilities.URIUtilities;
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
```
Comment (SPalominos) [inline h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java:52]: Missing class description.

File: h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+ * @author Erwan Bocher, CNRS, 2020
+ */
```
Comment (SPalominos) [inline h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java:54]: If the code is coming from Orbisdata, author missing.

File: h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java (hunk 2 above)
Comment (ebocher) [reply h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java:58]: Fixed my bad ;(

File: h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java (hunk 3 above)
Comment (ebocher) [reply h2gis-functions/src/main/java/org/h2gis/functions/io/utility/IOMethods.java:62]: Fixed my bad ;(

**Review by SPalominos — APPROVED** [review]

**Comment by nicolas-f** [comment]
Hello.

Its just a feedback on software design.

Maybe I would like to use IOMethods for my software by creating an instance of it, then pushing additional( or customized) drivers not in H2GIS. IOMethods would use https://github.com/orbisgis/h2gis/blob/master/h2gis-api/src/main/java/org/h2gis/api/DriverFunction.java#L59 instead of the current switch. So IOMethods would contain a non-static array of custom FileDrivers.

For reference, the manager of File Drivers was https://github.com/orbisgis/orbisgis/blob/master/bundles/dbjobs/src/main/java/org/orbisgis/dbjobs/service/DriverFunctionContainerImpl.java in OrbisGIS.

**Comment by ebocher** [comment]
Good idea. Feel free to send a patch

**Comment by nicolas-f** [comment]
> Good idea. Feel free to send a patch
> We will wait before merging this PR

ok I'm on it

**Comment by ebocher** [comment]
@nicolas-f 
Is there a channel to contact you ?

**Pull request merged**